frontend/src/pages/main/topup/confirm/TopUpConfirmation.tsx

```diff
@@ -0,0 +1,108 @@
+import {TopUpPlaceholder} from "../../../../common/constansts/placeholders";
+
+const initialCardData: ICard = {
+    cardName: "",
+    cardNumber: 0,
+    expirationDate: "",
+    ccv: 0
+}
+
+const initialWalletTopUp: IWalletTopUp = {
+    amount: 0
+}
+
+const TopUpConfirmation: React.FC<TopUpConfirmationProps> = ({walletId}) => {
+
+    const [cardData, setCardData] = useState<ICard>(initialCardData);
+    const [sendNotification, setSendNotification] = useState<boolean>(false);
+    const [walletTopUp, setWalletTopUp] = useState<IWalletTopUp>(initialWalletTopUp);
+    const [isAmountValid, setIsAmountValid] = useState<boolean>(false);
+    const navigate = useNavigate();
+    const dispatch = useAppDispatch();
+
+    const updateAmountChange = (amount: number) => {
+        setWalletTopUp({...walletTopUp, amount});
+    };
+
+    const updateIsAmountValidChange = (isAmountValid: boolean) => {
+        setIsAmountValid(isAmountValid);
+    };
+
+    const updateCardField = (data: string | number, cardFieldType: CardFieldType) => {
+        switch (cardFieldType) {
+            case CardFieldType.CARD_NAME:
+                setCardData({...cardData, cardName: data as string});
+                break;
+            case CardFieldType.CARD_NUMBER:
+                setCardData({...cardData, cardNumber: data as number});
+                break;
+            case CardFieldType.CARD_EXPIRATION_DATE:
+                setCardData({...cardData, expirationDate: data as string});
+                break;
+            case CardFieldType.CARD_CCV:
+                setCardData({...cardData, ccv: data as number});
+                break;
+            default:
+                break;
+        }
+
+    }
+
+    const handleNotification = () => {
+        setSendNotification(!sendNotification);
+    }
+
+    const handleCancel = () => {
+        navigate(Paths.ANY);
+    }
+
+    const handleConfirm = async () => {
+        const receivedWallet = await topUp(walletId, walletTopUp);
+        if (!receivedWallet) return;
+        dispatch(showNotification({show: true, type: NotificationType.SUCCESS, message:Notifications.SUCCESSFUL_TOP_UP}));
+        navigate(Paths.ANY);
+    }
+
+    return (
+        <Card className="card-background card-payment">
+            <CardContent className="card-confirm-content">
+                <CardTitle title={Titles.CARD_INFORMATION}/>
+                <CardInformationField cardFieldType={CardFieldType.CARD_NAME} updateField={updateCardField} placeHolder={TopUpPlaceholder.CARD_NAME} isFullWidth={true}/>
+                <CardInformationField cardFieldType={CardFieldType.CARD_NUMBER} updateField={updateCardField} placeHolder={TopUpPlaceholder.CARD_NUMBER} isFullWidth={true}/>
+                <Box className="privacy-data-card">
+                    <CardInformationField cardFieldType={CardFieldType.CARD_EXPIRATION_DATE} updateField={updateCardField} placeHolder={TopUpPlaceholder.EXPIRATION_DATE} isFullWidth={false}/>
```

Comment:
Maybe it can be useful. At least, it's easier in terms of validation, since you don't need to work with string
https://mui.com/x/react-date-pickers/date-picker/#views
https://stackoverflow.com/questions/64112889/month-and-year-picker-in-material-ui-react